test/unit/cmap/command_monitoring_events.test.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const { Msg, Query, GetMore, KillCursor } = require('../../../src/cmap/commands');
+const { CommandStartedEvent } = require('../../../src/cmap/command_monitoring_events');
+const { expect } = require('chai');
+const { Long } = require('bson');
+
+describe('Command Monitoring Events - unit/cmap', function () {
+  const commands = [
+    new Query('admin.$cmd', { a: { b: 10 }, $query: { b: 10 } }, {}),
+    new Query('hello', { a: { b: 10 }, $query: { b: 10 } }, {}),
+    new Msg('admin.$cmd', { b: { c: 20 } }, {}),
+    new Msg('hello', { b: { c: 20 } }, {}),
+    new GetMore('admin.$cmd', Long.fromNumber(10)),
+    new GetMore('hello', Long.fromNumber(10)),
+    new KillCursor('admin.$cmd', [Long.fromNumber(100), Long.fromNumber(200)]),
+    new KillCursor('hello', [Long.fromNumber(100), Long.fromNumber(200)]),
+    { ns: 'admin.$cmd', query: { $query: { a: 16 } } },
+    { ns: 'hello there', f1: { h: { a: 52, b: { c: 10, d: [1, 2, 3, 5] } } } }
+  ];
+
+  commands.forEach(c => {
```

Comment:
tiniest nit: can we use for-of loop here, it's completely stylistic but its friendlier when debugging.